go/chat/boxer.go

```diff
@@ -598,28 +589,47 @@ func (b *Boxer) verifyMessageHeaderV1(ctx context.Context, header chat1.HeaderPl
 		return verifyMessageRes{}, NewPermanentUnboxingError(libkb.NoKeyError{Msg: "key invalid for sender at message ctime"})
 	}
 
+	// check signature
+	hcopy := header
+	hcopy.HeaderSignature = nil
+	hpack, err := b.marshal(hcopy)
+	if err != nil {
+		return verifyMessageRes{}, NewPermanentUnboxingError(err)
+	}
+	if !b.verify(verifyKey, hpack, *header.HeaderSignature, libkb.SignaturePrefixChat) {
+		return verifyMessageRes{}, NewPermanentUnboxingError(libkb.BadSigError{E: "header signature invalid"})
+	}
+
 	return verifyMessageRes{
 		senderDeviceRevokedAt: revoked,
 	}, nil
 }
 
 // verify verifies the signature of data using SignatureInfo.
-func (b *Boxer) verify(data []byte, si chat1.SignatureInfo, prefix libkb.SignaturePrefix) bool {
+func (b *Boxer) verify(verifyKey keybase1.BinaryKID, data []byte, si chat1.SignatureInfo, prefix libkb.SignaturePrefix) bool {
 	sigInfo := libkb.NaclSigInfo{
 		Version: si.V,
 		Prefix:  prefix,
```

Comment:
Unrelated to the current diff, but: Passing around the prefix here feels wrong to me. Applications are supposed to hardcode the prefix (as we're doing, correctly). If anyone verifies a siginfo bundle without overwriting this field in it first, it totally defeats the purpose of having a prefix. It's a footgun.